Lib/socket.py

```diff
@@ -360,7 +360,7 @@ def _sendfile_use_sendfile(self, file, offset=0, count=None):
             if not fsize:
                 return 0  # empty file
             # Truncate to 1GiB to avoid OverflowError, see bpo-38319.
-            blocksize = min(count or fsize, 2 ** 30)
+            blocksize = min(count or fsize - offset, 2 ** 30)
```

Comment:
What if `offset >= fsize`?

Please add tests for `offset == fsize` and `offset > fsize`.